SDK/AppCenter/Microsoft.AppCenter.Shared/AppCenter.cs

```diff
@@ -18,6 +18,13 @@ public partial class AppCenter
         const string TargetKeyName = "target";
         const string TargetKeyNameUpper = "Target";
         const string AppSecretKeyName = "appsecret";
+        const string SecretsPattern = @"([^;=]+)=([^;=]+);?";
+
+#if NETSTANDARD
+        static readonly Regex _secretsRegex = new Regex(SecretsPattern);
+#else
+        static readonly Regex _secretsRegex = new Regex(SecretsPattern, RegexOptions.Compiled);
```

Comment:
What is the reason to use `RegexOptions.Compiled` option when it's not NET Standard?

https://docs.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-options?view=netframework-4.8#compiled-regular-expressions
> By default, regular expressions in .NET are interpreted. When a Regex object is instantiated or a static Regex method is called, the regular expression pattern is parsed into a set of custom opcodes, and an interpreter uses these opcodes to run the regular expression. This involves a tradeoff: The cost of initializing the regular expression engine is minimized at the expense of run-time performance.

Performance-wise in C#, comparing interpreted regex to compiled regex, interpreted regex saves time at initialization but less efficient at execution time. C# documentation recommends using interpreted regex (default) when used infrequently.

https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices#interpreted-vs-compiled-regular-expressions
> we recommend that you use interpreted regular expressions when you call regular expression methods with a specific regular expression relatively infrequently. You should use compiled regular expressions when you call regular expression methods with a specific regular expression relatively frequently.

In this case, regex is used to parse app secrets string which is only called once per app lifecycle which is very infrequent, therefore from a perf standpoint, interpreted regex would be a more reasonable choice.

Reply:
.NET Standard 1.0 does not have this option.
But the code never runs on .NET standard anyway, we bait and switch the real implementation.

Reply:
The "expected" usage pattern of `AppCenter.Start` is to be called at app startup time once per lifecycle (infrequent), though there is nothing stops developers to call them on every `onResume` (frequent). The compiled only has advantage when the regex is called frequently which is not the "mainstream" way to consume this API.

From what I read online, the perf difference of interpreted regex and compiled regex in C# is probably not significant no matter which one we choose. But to know it for sure, the only way is to benchmark testing it using StopWatch--I don't think it's worth it. 

I slightly disagree but not a big deal.

Reply:
We document to add AppCenter.start in Xamarin Forms onStart() which happens multiple times per process on Android (for example back button and start again).

Reply:
That's fair. Thanks.